src/TensorFlowNET.Core/Binding.cs

```diff
@@ -10,7 +10,13 @@ public static partial class Binding
 
         /// <summary>
         ///     Alias to null, similar to python's None.
+        ///     For TensorShape, please use Unknown
         /// </summary>
         public static readonly object None = null;
+
+        /// <summary>
+        /// Used for TensorShape None
+        /// </summary>
+        public static readonly int Unknown = -1;
```

Comment:
This is a horrible solution, just make None = -1, users of the library will never guess how to use this nor will try to dig that deep for examples.